Assets/_BForBoss/Scripts/SpeedometerBehaviour.cs

```diff
@@ -0,0 +1,55 @@
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace BForBoss
+{
+    public interface ICharacterSpeed
+    {
+        float Speed { get; }
+    }
+    
+    public class SpeedometerBehaviour : MonoBehaviour
+    {
+        [SerializeField] private Image _meter = null;
+        [SerializeField] private TMP_Text _speedLabel = null;
+        
+        private ICharacterSpeed _characterSpeed = null;
+        private float _maxSpeed = 50f;
+        
+        public void Initialize(ICharacterSpeed characterSpeed)
+        {
+            _characterSpeed = characterSpeed;
+        }
+
+
+        // Placeholder
+        private void Awake()
+        {
+            var character = FindObjectOfType<FirstPersonPlayer>();
+            Initialize(character);
+        }
+
+        private void Update()
+        {
+            if (_characterSpeed == null)
+            {
+                return;
+            }
+            
+            SetMeter();
+            SetSpeedLabel();
+        }
+
+        private void SetMeter()
+        {
+            var percentage = _characterSpeed.Speed / _maxSpeed;
```

Comment:
if you do end up setting the _maxSpeed variable to be editable via insepctor, make sure it has the `[Min]` attribute (set to at least 1f) so that this value is never 0 and cause an exception

Reply:
yup for sure